dirwatcher.py

```diff
@@ -0,0 +1,204 @@
+        for path, value in dict_.items():
+            line_count = value
+            with open(path) as f:
+                for line in f.readlines()[line_count:]:
+                    line_count += 1
+                    if text in line:
+                        logger.info(
+                            f'Found "{text}" in "{path}" (Line {line_count})')
+            dict_[path] = line_count
+        return dict_
+
+    else:
+        logger.info('Directory has no searchable files!')
+
+
+def compare(state=state, dict=current_dict):
+    '''compares state with current_dir and logs any changes'''
+    if len(state) < len(current_dict):
+        for path in current_dict:
+            if path not in state:
+                logger.info(f'"{path}" added to directory.')
+    if len(state) > len(current_dict):
+        for path in state:
+            if path not in current_dict:
+                logger.info(f'"{path}" removed from directory.')
+
+
+def create_parser():
+    '''creates parser that defines acceptable options flags (args)'''
+    parser = argparse.ArgumentParser(
+        description="Program searches directory files for indicated text.")
+    parser.add_argument(
+        'directory', help="Provide a directory path as a string. The program"
+        "will parse files in the directory for the search_text provided.",
+        type=str)
+    parser.add_argument(
+        'search_text',
+        help="Provide a text string to search directory files."
+        " If found, program will return filename and line number of text's"
+        "location.", type=str)
+    parser.add_argument(
+        '--interval', '-i', help="Provide an integer (in seconds).  "
+        "Program will re-run itself once per second(s) indicated.", type=int)
+    parser.add_argument(
+        '--extension', '-e',
+        help="Provide a string text extension (e.g. '.pdf')"
+        ".  Program will filter query to only search for"
+        "indicated file types.", type=str)
+    return parser
+
+
+def log_banner(msg, name, time, end=''):
+    '''create a log banner that executes at program startup and shutdown'''
+    text = ('----------------------------------------------------------\n'
+            f'{time} - {msg} PROGRAM {name}\n{end}'
+            '----------------------------------------------------------\n')
+    with open('dirwatcher.log', 'a') as f:
+        f.write(text)
+
+
+def main():
+    '''runs progam and polls continuously and logs info/errors
+    unless signaled to shutdown'''
+
+    timer = time.time()
+    start_formatted = datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+    log_banner('RUNNING', ''.join(sys.argv[:1]).upper(), start_formatted)
+
+    # parse sys args passed
+    parser = create_parser()
+    args = sys.argv[1:]
+    ns = parser.parse_args(args)
+
+    # exit program if signal received
+    signal.signal(signal.SIGTERM, action_signal)
+    signal.signal(signal.SIGINT, action_signal)
+
+    while not close_program:
```

Comment:
We should review the logic that is going on within this outer while-loop.  I think you need to restructure your if-statements.  
What will happen if the target directory to watch, does not yet exist?  It will log an error, and print_usage, and then what next?  It tries to watch the directory anyway, and gets into a very tight loop of errors and it fills up the log files quickly.